docs/coordinates/index.rst

```diff
@@ -425,6 +425,60 @@ and transforming velocities. These are available both via the lower-level
 For more details on velocity support (and limitations), see the
 :doc:`velocities` page.
 
+.. _astropy-coordinates-masks:
+
+Masks
+-----
+
+Sometimes you may have incomplete information about objects, e.g., some have
+distances while others have not. `~astropy.coordinates` supports using masks
+for such purposes, using the |Masked| class::
+
+    >>> from astropy.utils.masked import Masked
+    >>> distance = Masked([0.1, np.nan]*u.kpc, mask=[False, True])
+    >>> sc = SkyCoord([1., 2.]*u.hourangle, [3., 4.]*u.deg, distance=distance)
+    >>> sc
+    <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, kpc)
+        [(15., 3., 0.1), (30., 4., ———)]>
+
+The masks propagates as you would expect::
+
+    >>> sc.separation(sc[0])  # doctest: +FLOAT_CMP
+    <MaskedAngle [ 0.        , 15.00502838] deg>
+    >>> sc.separation_3d(sc[0])  # doctest: +FLOAT_CMP
+    <MaskedDistance [ 0., ———] kpc>
+    >>> from warnings import catch_warnings
```

Comment:
The change to GCRS sadly means the `nan` give runtime warnings, which, with our default settings, cause test failures. If anyone knows a more elegant solution to have the tests pass without littering the documentation with this, please let me know... (in `io.fits`, tests are skipped for this reason, but don't really want to do that here).

Reply:
Because you mention the warnings in the text then it would be best to [declare them as part of the intended output and show them](https://github.com/scientific-python/pytest-doctestplus?tab=readme-ov-file#showing-warnings).

Reply:
Thanks, that's super helpful!